lib/cli/do_volapi.js

```diff
@@ -0,0 +1,393 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright (c) 2016, Joyent, Inc.
+ */
+
+/*
+ * The 'sdcadm experimental volapi' CLI subcommand.
+ */
+
+var assert = require('assert-plus');
+var util = require('util'),
+    format = util.format;
+var vasync = require('vasync');
+
+var common = require('../common');
+var errors = require('../errors');
+var DownloadImages = require('../procedures/download-images').DownloadImages;
+var shared = require('../procedures/shared');
+
+var NB_MBS_IN_GB = 1024;
+var NFS_SHARED_VOLUMES_PACKAGES_NAME_PREFIX = 'nfs_shared_volume_';
```

Comment:
FWIW, I'd prefer that names for things in the volumes world start with "vol" or "volume" instead of using "nfs". Nfs is mostly an implementation detail here, isn't it? Perhaps likewise for the VM.internal_metadata.nfsvolumes field you mentioned earlier.